### PR TITLE
Improve Lightning package for external use

### PR DESCRIPTION
**Pull request by mlxd**
**Context:** To allow Lightning to be used in external projects, and support compilation through Spack, we require some adaptions to the external package management, and how certain files are supported in targets. This PR allows better support for Lightning as an external library, usable with and without Python, and enabling inclusion of dependencies when available.

**Description of the Change:** Update to CMake targets, build variant supports, and Kokkos package inclusion.

**Benefits:** Improves extensibility of Lightning, and allows support for Spack.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: cmake/process_options.cmake [hunk 1]
```diff
@@ -141,7 +143,62 @@ if(ENABLE_KOKKOS)
...
+        add_library( kokkoskernels SHARED IMPORTED GLOBAL)
+
+        cmake_path(GET kokkos_core_inc ROOT_PATH kokkos_INC_DIR)
+        cmake_path(GET kokkos_sparse_inc ROOT_PATH kokkoskernels_INC_DIR)
+
+        set_target_properties( kokkos PROPERTIES IMPORTED_LOCATION ${kokkos_core_lib})
+        set_target_properties( kokkoskernels PROPERTIES IMPORTED_LOCATION ${kokkos_kernels_lib})
+        set_target_properties( kokkos PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${kokkos_INC_DIR}")
+        set_target_properties( kokkoskernels PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${kokkoskernels_INC_DIR}")
+        target_link_libraries( lightning_external_libs INTERFACE kokkos kokkoskernels)
```
Comment (AmintorDusko) [inline cmake/process_options.cmake:198]: In the case of finding Kokkos, this linking step will only work if Kokkos was found on the second try when looking for the files.

File: cmake/process_options.cmake [hunk 2]
```diff
@@ -172,10 +229,10 @@ if(ENABLE_KOKKOS)
 
         get_target_property(kokkoskernels_INC_DIR kokkoskernels INTERFACE_INCLUDE_DIRECTORIES)
         set_target_properties(kokkoskernels PROPERTIES INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${kokkoskernels_INC_DIR}")
-
+        target_link_libraries(lightning_external_libs INTERFACE kokkos kokkoskernels)
     endif()
     target_compile_options(lightning_compile_options INTERFACE "-D_ENABLE_KOKKOS=1")
-    target_link_libraries(lightning_external_libs INTERFACE Kokkos::kokkos Kokkos::kokkoskernels)
```
Comment (AmintorDusko) [inline cmake/process_options.cmake:178]: I think we would need to link like this if Kokkos was found with find_package.

File: cmake/process_options.cmake [hunk 3]
```diff
@@ -8,6 +8,132 @@
...
 include_guard()
 
+##############################################################################
+
+# Macro to aid in finding Kokkos with 3 potential install options:
+# 1. Fully integrated Kokkos packages and CMake module files
+# 2. Statically compiled libraries and headers
+# 3. Not installed, so fall back to building from source.
+
+macro(FindKokkos target_name)
```
Comment (AmintorDusko) [inline cmake/process_options.cmake:18]: This macro was nicely done!

File: pennylane_lightning/src/simulator/CMakeLists.txt [hunk 4]
```diff
@@ -15,5 +15,5 @@ target_link_libraries(lightning_simulator PRIVATE lightning_compile_options
                                                   lightning_external_libs
                                                   lightning_gates
                                                   lightning_utils)
-
+#target_link_libraries(lightning_simulator PUBLIC lightning_external_libs)
```
Comment (AmintorDusko) [inline pennylane_lightning/src/simulator/CMakeLists.txt:18]:
```suggestion
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -76,11 +83,13 @@ add_library(pennylane_lightning INTERFACE)
...
                                                         lightning_simulator
-                                                        lightning_algorithms)
+                                                        lightning_algorithms
+                                                        lightning_gates
+)
 target_include_directories(pennylane_lightning INTERFACE "pennylane_lightning/src")
 
 #####################################################
-
+if(ENABLE_PYTHON)
```
[the remark this answers is not in the document]
Comment (mlxd) [reply CMakeLists.txt:92]: Sure thing.

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -116,7 +114,7 @@ def circuit():
 
 This release contains contributions from (in alphabetical order):
 
-Amintor Dusko, Christina Lee, Chae-Yeun Park
+Amintor Dusko, Christina Lee, Lee J. O'Riordan, Chae-Yeun Park
```
[the remark this answers is not in the document]
Comment (mlxd) [reply .github/CHANGELOG.md:117]: Well that was a dodgy merge. Yep will fix it.

**Review by AmintorDusko — APPROVED** [review]
Thank you for the nice job!

**Comment by multiphaseCFD** [comment]
Hey @mlxd, nice work!

**Review by multiphaseCFD — APPROVED** [review]

**Pull request merged**